### PR TITLE
Add wayland compositor

### PR DESCRIPTION
**Pull request by neithernut**
Add preliminary implementation of the wayland compositor.
Stil WIP.

### CODE REVIEW

**Comment by neithernut** [comment]
Target #107.

**Comment by neithernut** [comment]
I'm done here. What's missing is actually creating a surface and assigning it to an output. That will be done in other PRs.

File: src/compositor/wayland_compositor.c [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+// wayland-server.h has to be included before wayland-server-protocol.h
+#include <wayland-server.h>
+#include <wayland-server-protocol.h>
+
+#include "compositor/wayland_compositor.h"
+#include "objects/wayland_obj.h"
+#include "util/arithmetical.h"
+#include "util/wayland.h"
+
+#define WAYLAND_COMPOSITOR_VERSION  (3)
```
Comment (matthiasbeyer) [inline src/compositor/wayland_compositor.c:42]: Also here, please add a note what this is for.

File: src/compositor/wayland_compositor.c [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+ws_wayland_compositor_init(void) {
+    static bool is_init = false;
+    if (is_init) {
+        return 0;
+    }
+
+    // get the display
+    struct wl_display* display = ws_wayland_acquire_display();
+    if (!display) {
+        return -1;
```
Comment (matthiasbeyer) [inline src/compositor/wayland_compositor.c:123]: Why not an errno const?

Comment (neithernut) [inline src/compositor/wayland_compositor.c:123]: There are too many possibilities why this could fail.

Comment (matthiasbeyer) [inline src/compositor/wayland_compositor.c:123]: Then, please, add a note which says exactly this.

**Comment by neithernut** [comment]
Any more annotations?

**Comment by matthiasbeyer** [comment]
Just one: ACK.

**Pull request merged**